### PR TITLE
Refactor Google::Gax::Stub::new

### PR DESCRIPTION
**Pull request by blowmage**
Use a more familiar convention for creating a Gax Stub object. Also, simplify the credentials arguments and have the Gax wrapper sort the various types of credentials values out.

The GAPIC generator side of this change is in: googleapis/gapic-generator-ruby#116

### CODE REVIEW

File: lib/google/gax/api_call.rb [hunk 1]
```diff
@@ -98,9 +98,9 @@ def initialize stub_method
       #   echo_channel = GRPC::Core::Channel.new(
       #     "localhost:7469", nil, :this_channel_is_insecure
       #   )
-      #   echo_stub = Google::Gax::Grpc.create_stub(
-      #     "localhost", 7469, channel: echo_channel,
-      #     &Google::Showcase::V1alpha3::Echo::Stub.method(:new)
+      #   echo_stub = Google::Gax::Grpc::Stub.new(
+      #     Google::Showcase::V1alpha3::Echo::Stub.method(:new)
```
Comment (jbolinger) [inline lib/google/gax/api_call.rb]: Should this be `Google::Showcase::V1alpha3::Echo::Stub` now?

File: lib/google/longrunning/operations_client.rb [hunk 2]
```diff
@@ -299,31 +299,13 @@ def delete_operation request = nil, options: nil, **request_fields, &block
...
-        port = self.class::DEFAULT_SERVICE_PORT
-        interceptors = self.class::GRPC_INTERCEPTORS
-        Google::Gax::Grpc.create_stub(
-          service_path,
-          port,
-          chan_creds:   chan_creds,
-          channel:      channel,
-          updater_proc: updater_proc,
+        Google::Gax::Grpc::Stub.new(
+          Google::Longrunning::Operations::Stub.method(:new),
```
Comment (jbolinger) [inline lib/google/longrunning/operations_client.rb]: Isn't it `Google::Longrunning::Operations::Stub` now?

**Comment by blowmage** [comment]
I pushed the create PR button before this was ready to be reviewed. PTAL now.

**Comment by jbolinger** [comment]
@blowmage I think you still need to update one of the tests to use the new method signature.

**Comment by blowmage** [comment]
@jbolinger I was working off some old commits. Updated the branch to the commit with tests. CI is passing now. PTAL.

**Pull request merged**